packages/web-app-files/src/helpers/resource/actions/transfer.ts

```diff
@@ -0,0 +1,182 @@
+    private targetSpace: SpaceResource,
+    private targetFolder: Resource,
+    private clientService: ClientService,
+    createModal: (modal: object) => void,
+    hideModal: () => void,
+    showMessage: (data: object) => void,
+    $gettext: (msg: string) => string,
+    $ngettext: (msgid: string, plural: string, n: number) => string,
+    $gettextInterpolate: (msg: string, context: object, disableHtmlEscaping?: boolean) => string
+  ) {
+    super(createModal, hideModal, showMessage, $gettext, $ngettext, $gettextInterpolate)
+  }
+
+  hasRecursion(): boolean {
+    if (this.sourceSpace.id !== this.targetSpace.id) {
+      return false
+    }
+    return this.resourcesToMove.some(
+      (resource: Resource) => this.targetFolder.path === resource.path
+    )
+  }
+
+  showRecursionErrorMessage() {
+    const count = this.resourcesToMove.length
+    const title = this.$ngettext(
+      "You can't paste the selected file at this location because you can't paste an item into itself.",
+      "You can't paste the selected files at this location because you can't paste an item into itself.",
+      count
+    )
+    this.showMessage({ title, status: 'danger' })
+  }
+
+  showResultMessage(errors, movedResources: Array<Resource>, transferType: TransferType) {
+    if (errors.length === 0) {
+      const count = movedResources.length
+      const ntitle =
+        transferType === TransferType.COPY
+          ? this.$ngettext(
+              '%{count} item was copied successfully',
+              '%{count} items were copied successfully',
+              count
+            )
+          : this.$ngettext(
+              '%{count} item was moved successfully',
+              '%{count} items were moved successfully',
+              count
+            )
+      const title = this.$gettextInterpolate(ntitle, { count }, true)
+      this.showMessage({
+        title,
+        status: 'success'
+      })
+      return
+    }
+    let title = this.$gettextInterpolate(
+      transferType === TransferType.COPY
+        ? this.$gettext('Failed to copy %{count} resources')
+        : this.$gettext('Failed to move %{count} resources'),
+      { count: errors.length },
+      true
+    )
+    if (errors.length === 1) {
+      title = this.$gettextInterpolate(
+        transferType === TransferType.COPY
+          ? this.$gettext('Failed to copy "%{name}"')
+          : this.$gettext('Failed to move "%{name}"'),
+        { name: errors[0]?.resourceName },
+        true
+      )
+    }
+    this.showMessage({
+      title,
+      status: 'danger'
+    })
+  }
+
+  async perform(transferType: TransferType): Promise<Resource[]> {
+    if (this.hasRecursion()) {
+      this.showRecursionErrorMessage()
+      return []
+    }
+    if (this.sourceSpace.id !== this.targetSpace.id && transferType === TransferType.MOVE) {
+      const doCopyInsteadOfMove = await this.resolveDoCopyInsteadOfMoveForSpaces()
+      if (!doCopyInsteadOfMove) return []
```

Comment:
In general we don't have single line ifs in the code base. Please convert to
```
if (!doCopyInsteadOfMove) {
  return []
}
```
to keep the code style consistent.

Reply:
(we should introduce an eslint rule for  this)